### PR TITLE
feat(PPDSC-2077): add grid-auto props and stylePreset overrides to GridLayout

### PR DESCRIPTION
**Pull request by Xin00163**
PPDSC-2077

- add grid-auto-columns and grid-auto-rows -> [A Complete Guide to Grid | CSS-Tricks](https://css-tricks.com/snippets/css/complete-guide-grid/#aa-grid-auto-flow)  → prop name autoColumns and autoRows
- add grid-auto-flow -> [A Complete Guide to Grid | CSS-Tricks](https://css-tricks.com/snippets/css/complete-guide-grid/#aa-grid-auto-flow) → prop name autoFlow

**I have done:**
- [x] Written unit tests against changes
- [ ] Written functional tests against the component and/or NewsKit site
- [x] Updated relevant documentation

**I have tested manually:**
- [ ] The feature's functionality is working as expected on Chrome, Firefox, Safari and Edge
- [ ] The screen reader reads and flows through the elements as expected.
- [ ] There are no new errors in the browser console coming from this PR.
- [ ] When visual test is not added, it renders correctly on different browsers and mobile viewports (Safari, Firefox, small mobile viewport, tablet)
- [ ] The Playground feature is working as expected

**Before:**

**After:**

**Who should review this PR:**

**How to test:**

### CODE REVIEW

**Comment by newskit-engineering** [comment]
You can preview these changes on:
  Documentation Site: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-2077-add-grid-auto-and-stylepreset-to-gridlayout
  Storybook: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-2077-add-grid-auto-and-stylepreset-to-gridlayout/storybook

File: scripts/templates-integration-test/next.config.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
```
Comment (evgenitsn) [inline scripts/templates-integration-test/next.config.js:1]: Nice find! On my branch I've updated this file and added the `cp ./scripts/templates-integration-test/next.config.js ./newskit-integration-test/` line in `config.yml` and it fixed the issue. Do you think we need the rest of the changes that you've added to the `config.yml`?
I will mark them as separate comments.

File: .circleci/config.yml [hunk 2]
```diff
@@ -1031,21 +1031,22 @@ jobs:
       - run:
           name: Install yalc and create-next-app
           command: |
-            yarn global add yalc && yarn add create-next-app@latest
+            yarn global add yalc
```
Comment (evgenitsn) [inline .circleci/config.yml:1034]: This one

(See my other comment)

File: .circleci/config.yml [hunk 3]
```diff
@@ -1031,21 +1031,22 @@ jobs:
...
       - run:
           name: Yalc publish and run create-nextjs-app
           command: |
             cd dist
             yalc publish
             cd ..
-            npx create-next-app@latest newskit-integration-test
+            yarn create next-app newskit-integration-test
```
Comment (evgenitsn) [inline .circleci/config.yml:1041]: And this one

(See my other comment)

File: src/grid-layout/styled.ts [hunk 4]
```diff
@@ -59,6 +67,27 @@ export const StyledGridLayout = styled.div<GridLayoutProps>`
     gridTemplateAreas: areas,
   }))};
 
+  ${handleResponsiveProp(
+    {autoFlow: GRID_DEFAULT_PROPS.autoFlow},
+    ({autoFlow}, {theme}) => ({
+      gridAutoFlow: mapTemplate(theme, autoFlow),
```
Comment (mutebg) [inline src/grid-layout/styled.ts:73]: Do you need `mapTemplate` here? 
autoFlow is not sizing token

File: .circleci/config.yml (hunk 2 above)
Comment (Xin00163) [reply .circleci/config.yml:1034]: I did this to clean it up. The official docs says to do `yarn create next-app` and I think this is redundant

File: .circleci/config.yml (hunk 3 above)
Comment (Xin00163) [reply .circleci/config.yml:1041]: We shouldn't mix npm with yarn. This will trigger a warning.

File: src/grid-layout/styled.ts (hunk 4 above)
Comment (Xin00163) [reply src/grid-layout/styled.ts:73]: You are right. I will update

File: src/grid-layout/styled.ts [hunk 5]
```diff
@@ -59,6 +67,27 @@ export const StyledGridLayout = styled.div<GridLayoutProps>`
...
+    {autoFlow: GRID_DEFAULT_PROPS.autoFlow},
+    ({autoFlow}) => ({
+      gridAutoFlow: autoFlow,
+    }),
+  )}
+
+  ${handleResponsiveProp(
+    {autoRows: GRID_DEFAULT_PROPS.autoRows},
+    ({autoRows}) => ({
+      gridAutoRows: autoRows,
```
Comment (mutebg) [inline src/grid-layout/styled.ts:80]: IMO gridAutoRows and gridAutoColumns can be size token, but not autoFlow

**Review by LukeFinch — APPROVED** [review]
Nice work, looks good!

File: src/grid-layout/styled.ts (hunk 5 above)
Comment (Xin00163) [reply src/grid-layout/styled.ts]: Sorry. Have updated

**Review by mutebg — APPROVED** [review]

**Review by LukeFinch — APPROVED** [review]

**Comment by Xin00163** [comment]
After https://github.com/newscorp-ghfb/newskit/pull/419 is merged, applying stylePreset to gridLayout will result in styles being applied as Flag is now using GridLayout. This result in more than 300 snapshot changes. After chatting with @mutebg , we decided not to add stylePreset to GridLayout overrides for now.

**Review by mutebg — APPROVED** [review]

**Review by LukeFinch — APPROVED** [review]

**Pull request merged**